File: retort-annotations/src/main/lombok/co/kaioru/retort/annotation/adapter/AliasAnnotationProcessor.java
package co.kaioru.retort.annotation.adapter;

import co.kaioru.retort.ICommand;
import co.kaioru.retort.annotation.AbstractAnnotationProcessor;
import co.kaioru.retort.annotation.type.Alias;
import co.kaioru.retort.annotation.type.Aliases;

import java.lang.reflect.Method;

public class AliasAnnotationProcessor extends AbstractAnnotationProcessor {

    @Override
    public void process(ICommand command, Object object, Method method) {
        if (method.isAnnotationPresent(Aliases.class)) {
            for (Alias alias : method.getAnnotation(Aliases.class).value())
                command.registerAlias(alias.value());
        }
    }

}

File: retort-annotations/src/main/kotlin/co/kaioru/retort/annotation/impl/adapter/AliasAnnotationProcessor.kt
package co.kaioru.retort.annotation.impl.adapter

import co.kaioru.retort.ICommand
import co.kaioru.retort.annotation.Alias
import co.kaioru.retort.annotation.impl.AnnotationProcessor
import java.lang.reflect.Method

class AliasAnnotationProcessor<I, O> : AnnotationProcessor<I, O>() {
    override fun process(command: ICommand<I, O>, any: Any, method: Method) {
        method.getAnnotationsByType(Alias::class.java).forEach {
            command.registerAlias(it.value)
        }
    }
}